src/repart/repart.c

```diff
@@ -7552,7 +7571,37 @@ static int context_fstab(Context *context) {
                 }
         }
 
-        r = flink_tmpfile(f, t, path, 0);
+        fputs(AUTOMATIC_FSTAB_HEADER_END "\n", f);
+
+        switch (arg_append_fstab) {
+        case APPEND_AUTO:
+                r = read_full_file(path, &c, NULL);
+                if (r < 0) {
+                        if (r == -ENOENT) {
+                                log_debug("File fstab not found in %s", path);
+                                break;
+                        }
+                        return log_error_errno(r, "Failed to open %s: %m", path);
+                }
+
+                const char *acs, *ace;
+                acs = find_line(c, AUTOMATIC_FSTAB_HEADER_START);
+                if (acs) {
+                        fwrite(c, 1, acs - c, f);
+                        ace = find_line_after(acs, AUTOMATIC_FSTAB_HEADER_END);
+                        if (ace)
+                                fputs(ace, f);
+                } else
+                        fputs(c, f);
+                break;
```

Comment:
hmm, can you enclose this whole case block in a {}? the local variables `acs`and `ace` otherwise have such a weird lifetime (since they'll be allocated even in the APPEND_NO/APPEND_REPLACE cases)... i.e. just a an openeing `{` right after the `case APPEND_AUTO:` and a closing `}` right after the `break;` here.